### ISSUE TITLE
Make library OSGi ready

### ISSUE THREAD

**Issue by jludvice** [czgov/JAVA_ISDS#14]
It would be nice to make library OSGi ready to allow deployment to Apache Karaf.

**Comment by jludvice** [comment]
There might be issue with bouncycastle when deploying on JBoss Fuse
```
11:29:19,437 | INFO  | ead #2 - isds:// | DataBoxMessagesServiceImpl       | 301 - cz.abclinuxu.datoveschranky.isds - 1.0.0.SNAPSHOT | getListOfReceivedMessages: offset:1 limit:2147483647
11:29:19,513 | INFO  | ead #2 - isds:// | DataBoxMessagesServiceImpl       | 301 - cz.abclinuxu.datoveschranky.isds - 1.0.0.SNAPSHOT | getListOfReceivedMessages finished
11:29:19,513 | INFO  | ead #2 - isds:// | ISDSConsumer                     | 309 - com.github.czgov.isds.com.github.czgov.isds - 0.1.0.SNAPSHOT | Poll success, found 1 message envelopes.
11:29:19,514 | INFO  | ead #2 - isds:// | ISDSConsumer                     | 309 - com.github.czgov.isds.com.github.czgov.isds - 0.1.0.SNAPSHOT | Extracting headers of message 5285259.
11:29:19,514 | INFO  | ead #2 - isds:// | ISDSConsumer                     | 309 - com.github.czgov.isds.com.github.czgov.isds - 0.1.0.SNAPSHOT | Downloading message 5285259 in unmarshalled Message instance.
11:29:19,514 | INFO  | ead #2 - isds:// | DataBoxDownloadServiceImpl       | 301 - cz.abclinuxu.datoveschranky.isds - 1.0.0.SNAPSHOT | downloadMessage: id:5285259
11:29:19,769 | INFO  | ead #2 - isds:// | DataBoxDownloadServiceImpl       | 301 - cz.abclinuxu.datoveschranky.isds - 1.0.0.SNAPSHOT | downloadMessage successfull
11:29:19,769 | WARN  | ead #2 - isds:// | ISDSConsumer                     | 232 - org.apache.camel.camel-core - 2.17.0.redhat-630187 | Consumer Consumer[isds://?consumer.delay=5s&environment=test&password=xxxxxx&username=qzv4ej] failed polling endpoint: Endpoint[isds://?consumer.delay=5s&environment=test&password=xxxxxx&username=qzv4ej]. Will try again at next poll. Caused by: [java.lang.NoClassDefFoundError - Could not initialize class org.bouncycastle.cms.CMSSignedData]
java.lang.NoClassDefFoundError: Could not initialize class org.bouncycastle.cms.CMSSignedData
	at cz.abclinuxu.datoveschranky.impl.Validator.readTimeStamp(Validator.java:69)
	at cz.abclinuxu.datoveschranky.impl.MessageValidator.buildMessage(MessageValidator.java:318)
	at cz.abclinuxu.datoveschranky.impl.DataBoxDownloadServiceImpl.downloadMessage(DataBoxDownloadServiceImpl.java:46)
	at com.github.czgov.isds.ISDSConsumer.poll(ISDSConsumer.java:96)
	at org.apache.camel.impl.ScheduledPollConsumer.doRun(ScheduledPollConsumer.java:175)[232:org.apache.camel.camel-core:2.17.0.redhat-630187]
	at org.apache.camel.impl.ScheduledPollConsumer.run(ScheduledPollConsumer.java:102)[232:org.apache.camel.camel-core:2.17.0.redhat-630187]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_121]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)[:1.8.0_121]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)[:1.8.0_121]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)[:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_121]
```

**Comment by jludvice** [comment]
OSGi deployment tested on JBoss Fuse 6.3 GA.
It looks like bouncy castle version upgrade solved the class loading issue
`java.lang.NoClassDefFoundError: Could not initialize class org.bouncycastle.cms.CMSSignedData`

Also It was necesarry to widen `jax.xml.ws` import range to work with packages which are by default available in Fuse.


Installation of isds on Fuse would look like this:
**Note** Fuse should check maven central for release artifacts out of the box.
Snasphot artifacts must be explicitelly enabled in fuse config (config file `$FUSE_HOME/etc/org.ops4j.pax.url.mvn.cfg`).
```bash
# download features.xml from maven repo to fuse -- replace snapshot version for appropriate one
JBossFuse:karaf@root> features:addurl mvn:cz.abclinuxu.datoveschranky/karaf/1.0.0-SNAPSHOT/xml/features

# download jars and install them to fuse
JBossFuse:karaf@root> features:install isds
```

To verify everything is installed correctly:
```bash
JBossFuse:karaf@root> osgi:list
START LEVEL 100 , List Threshold: 50
   ID   State         Blueprint      Spring    Level  Name
# note that bcprov was already present in fuse
[ 293] [Active     ] [            ] [       ] [   80] JavaISDS :: Common (1.0.0.SNAPSHOT)
[ 294] [Active     ] [            ] [       ] [   80] JavaISDS :: JavaISDS (1.0.0.SNAPSHOT)
[ 295] [Active     ] [            ] [       ] [   80] JavaISDS :: Web Services (1.0.0.SNAPSHOT)
[ 296] [Active     ] [            ] [       ] [   80] bcpkix (1.54)
```

**Comment by jludvice** [comment]
resolved by #15
